doit/dependency.py

```diff
@@ -361,6 +338,96 @@ def remove_all(self):
         self._conn.execute('delete from doit')
 
 
+class BaseChecker(object):
+    """Base Checker that must be inherited."""
+
+    def __init__(self, backend):
+        self.backend = backend
+        self._set = self.backend.set
+        self._get = self.backend.get
+
+    def check_modified(self, file_path, state):
+        """Check if file in file_path is modified from previous "state".
+
+        @param file_path (string): file path
+        @param state (tuple), timestamp, size, md5
+        @returns (bool): True if dep is modified
+        """
+        pass
+
+    def save_state(self, task, dep):
+        """Save info after a task is successfuly executed"""
+        pass
+
+
+class DefaultChecker(BaseChecker):
+    """Default checker. Use the timestamp, size and md5sum."""
+
+    def check_modified(self, file_path, state):
+        try:
+            file_stat = os.stat(file_path)
```

Comment:
Calling os.stat() many times will cause a performance degradation for users who have thousands of small files...

Thats why I had the trouble to save os.stat() result and re-use it. Can you avoid it?